enhancements/insights/prometheus-alerts-for-insights-recommendations.md

```diff
@@ -0,0 +1,267 @@
+
+It is worth pointing out that if any key part of the Prometheus stack is not
+running as expected on the cluster, then this feature will not work. Prometheus
+manages both the metrics and the alerts, so without it, no alerts can be
+generated by IO.
+
+**Unhealthy Insights Operator scenario:**
+
+- If something goes wrong with IO, we would not want the Insights alerts to
+  remain hanging on the cluster indefinitely.
+- The alert should only fire if IO has been healthy at any point in the past 10
+  minutes. Healthy, in this context, means that there is no major error
+  happening (crashlooping, degraded operator status, etc.). A temporary error
+  state should not cause the alerts to be disabled.
+- If the user intentionally disables IO, the alerts should never fire. If IO is
+  disabled while some of the Insights alerts are firing, the alerts should
+  immediately stop firing (a part of their condition will be that IO is
+  enabled).
+- If IO was disabled by an accident, the user will be notified via the existing
+  alert.
+
+**Unhealthy Smart Proxy scenario:**
+
+- The latest successfully fetched and parsed report is too old
+  (`Meta.LastCheckedAt` is more than 8 hours ago)
+  - Reset the Insights recommendation metrics to avoid showing alerts about
+    recommendations that are no longer hitting.
+- Unable to fetch a report from Smart Proxy (endpoint unreachable, auth error, etc.)
+  - Log error, do not touch the Insights recommendation metrics.
+- The fetched report cannot be decoded into the expected structure, or the data
+  are otherwise invalid
+  - Log error, do not touch the Insights recommendation metrics.
+
+If there is anything wrong with the Smart Proxy or the reports returned by it,
+the metrics will not be modified until the validity of the last report expires.
+If IO is enabled, but something is wrong with the gathering or uploading, the
+metrics will not be directly affected for as long as the last report is valid.
+If IO has been disabled, or crashing for more then 10 minutes, the alerts will
+be disabled to avoid presenting the user with outdated information.
+
+### Risks and Mitigations
+
+- Disabling and re-enabling an alert (e.g, due to IO or Smart Proxy outage)
+  can trigger user-defined actions, even though there has not been any actual
+  change to the status of the recommendation. This could be very annoying if
+  it were to happen often or repeatedly over a short period of time.
+- The URL to detailed information about an active Insights recommendation will
+  be hard-coded into the alert template or the Insights Operator. If the
+  endpoint moves elsewhere, the URL will be broken and an updated version of the
+  alert template will have to be backported. The process can take several
+  months, and there is no guarantee that the user will perform the update.
+
+## Design Details
+
+### Open Questions [optional]
+
+1. Which operator conditions should fall under the unhealthy Insights Operator
+   state?
```

Comment:
Apart of the common clusteroperator conditions, we have these [special conditions](https://github.com/openshift/insights-operator/blob/master/pkg/controller/status/conditions.go#L8) so I think that whenever there's any condition related to a failure to obtain the new data (basically `UploadDegraded` and/or `InsightsDownloadDegraded`)

Reply:
The upload and download are certainly import to this functionality, and the `Disabled` status is definitely something we'd need to watch out for, but I'm not too sure about the `SCANotAvailabe`. Should that also be considered a fatal status, which should cause the alerts to be hidden?

Reply:
No I don't think so. `SCANotAvailable` is basically a situation when we can't download the SCA certs for some reason.

Reply:
@sdodson this might give you the most cause for concern:  if the insights operator can't reach the reporting service, it may report itself degraded, which would mean upgrades can't complete.  That sort of coupling to an external service could be fragile.